clang/include/clang/Sema/Sema.h

```diff
@@ -2075,6 +2075,10 @@ class Sema final : public SemaBase {
   /// Last section used with #pragma init_seg.
   StringLiteral *CurInitSeg;
   SourceLocation CurInitSegLoc;
+  /// Optional function name provided to #pragma init_seg([seg][, func-name]).
+  /// When present, stores the IdentifierInfo and its location.
+  IdentifierInfo *CurInitSegFn;
```

Comment:
Is there a reason that these and the above have to be separate?  I would expect us to only be parsing 1x at a time, right?  Could we combine all 4 into a single data structure?